### PR TITLE
fix: show full worktree name in sidebar badge

### PR DESCRIPTION
**Pull request by Pizzaface**
The worktree badge in the session sidebar was only capturing the first path segment after `.worktrees/` (e.g. just `fix`), because the regex used `[^/]+`. Since worktree directories mirror branch names like `fix/plan-mode-sandbox` (two segments), the badge was truncated.

**Fix:** Updated the regex to capture two segments: `[^/]+/[^/]+`, so the badge now shows the full worktree name (e.g. `fix/plan-mode-sandbox`).

### CODE REVIEW

**Comment by Pizzaface** [comment]
Re: worktree badge regex — Fixed. Replaced the two-segment regex with `indexOf` + `substring` that captures everything after `/.worktrees/`. This handles single-segment (`main`), two-segment (`fix/bar`), and deeper branch names (`feat/foo/bar`) correctly.

File: packages/ui/src/components/SessionSidebar.tsx [hunk 1]
```diff
@@ -1363,14 +1363,18 @@ export const SessionSidebar = React.memo(function SessionSidebar({
                                                             {/* Worktree badge — shown when cwd is inside a .worktrees/ directory */}
                                                             {(() => {
                                                                 const cwd = s.cwd || "";
-                                                                const match = cwd.match(/\/\.worktrees\/([^/]+)/);
-                                                                return match ? (
+                                                                const wtMarker = "/.worktrees/";
+                                                                const wtIdx = cwd.indexOf(wtMarker);
+                                                                const branchName = wtIdx !== -1
+                                                                    ? cwd.substring(wtIdx + wtMarker.length).replace(/\/$/, "")
```
[the remark this answers is not in the document]
Comment (Pizzaface) [reply packages/ui/src/components/SessionSidebar.tsx]: Fixed. Extracted a new `extractWorktreeName(cwd, knownCwds)` utility in `path.ts` that:

1. Checks known session cwds to find the longest worktree root prefix (same approach as `session-tree.ts`)
2. Falls back to first segment after `/.worktrees/` for single-segment branches

The badge now passes `liveSessions` cwds so slashed branch names like `feat/login` resolve correctly even when cwd is deeper (e.g. `/.worktrees/feat/login/src`).

11 new test cases added covering all edge cases.

File: packages/ui/src/components/SessionSidebar.tsx [hunk 2]
```diff
@@ -1363,14 +1363,15 @@ export const SessionSidebar = React.memo(function SessionSidebar({
                                                             {/* Worktree badge — shown when cwd is inside a .worktrees/ directory */}
                                                             {(() => {
                                                                 const cwd = s.cwd || "";
-                                                                const match = cwd.match(/\/\.worktrees\/([^/]+)/);
-                                                                return match ? (
+                                                                const knownCwds = liveSessions.map((ls) => ls.cwd || "");
+                                                                const branchName = extractWorktreeName(cwd, knownCwds);
```
[the remark this answers is not in the document]
Comment (Pizzaface) [reply packages/ui/src/components/SessionSidebar.tsx]: Fixed in e23f6d1. Added a `worktreeRoots()` helper that filters the cwd list to only the shortest path under each `/.worktrees/` prefix, then the call site in `SessionSidebar.tsx` uses it before passing to `extractWorktreeName`. Tests added for both the helper and the end-to-end case.

**Comment by Pizzaface** [comment]
@codex review

**Comment by Pizzaface** [comment]
@codex review

**Comment by Pizzaface** [comment]
@codex review

**Comment by Pizzaface** [comment]
@codex review
